cdap-proto/src/main/java/co/cask/cdap/proto/WorkflowTokenDetail.java

```diff
@@ -33,7 +33,7 @@
   private final Map<String, List<NodeValueDetail>> tokenData;
 
   public WorkflowTokenDetail(Map<String, List<NodeValueDetail>> tokenData) {
-    this.tokenData = ImmutableMap.copyOf(tokenData);
+    this.tokenData = Collections.unmodifiableMap(tokenData);
```

Comment:
Copy. Actually the original code is also wrong. It only copies the map, but not the `List`.